### DISCUSSION TITLE
Log output in Weblog

### DISCUSSION THREAD

**Discussion by CK4444** [HomeSpan/HomeSpan#320] [Ideas]
I just started to meddle around with HomeSpan and want to first of all express my big thanks to the developer and the community. 
Trying to get a rather simple script using the DHT22 working I wanted to make use of the WLog functionality (which is a great feature for itself). Soon I realized that only things are Weblogged which I define myself. I was hoping, and this would be the idea, that all the things being logged and printed in the serial monitor could be redirected to be logged in parallel at the Webserver. Especially, since I would not want by any means spoil the original HomeSpan Code with WLog Macros.

Could be a simple switch like: 
homeSpan.enableLogToWebLog()

Thanks, 
Christian

**Comment by HomeSpan** [comment]
The WebLog functionality is only meant for a limited number of information messages, whereas there are many, multi-line messages (mostly designed for debugging) that are output to the Serial monitor.  Having them all redirected to the WebLog mechanism would unfortunately not work well.   However, others have used are some alternatives whereby which seems to stream both Serial input and output to a web host so you can interact with CLI directly over the web.

**Reply by CK4444** [reply]
Thank you for the fast reply, I understand the concern.

**Comment by HomeSpan** [comment]
Also please see #275 re the DHT22.

**Reply by CK4444** [reply]
Thank you for the hint - always wondered why my DHT22 is working poorly every now and then ( on the ESP32)
